Generator/generator-botbuilder-java/generators/app/index.js

```diff
@@ -1,49 +1,268 @@
 'use strict';
+const pkg = require('../../package.json');
 const Generator = require('yeoman-generator');
-const chalk = require('chalk');
-const yosay = require('yosay');
 const path = require('path');
 const _ = require('lodash');
-const extend = require('deep-extend');
+const chalk = require('chalk');
 const mkdirp = require('mkdirp');
 
+const BOT_TEMPLATE_NAME_EMPTY = 'Empty Bot';
+const BOT_TEMPLATE_NAME_SIMPLE = 'Echo Bot';
+const BOT_TEMPLATE_NAME_CORE = 'Core Bot';
+
+const BOT_TEMPLATE_NOPROMPT_EMPTY = 'empty';
+const BOT_TEMPLATE_NOPROMPT_SIMPLE = 'echo';
+const BOT_TEMPLATE_NOPROMPT_CORE = 'core';
+
 module.exports = class extends Generator {
-  prompting() {
-    // Have Yeoman greet the user.
-    this.log(
-      yosay(`Welcome to the badass ${chalk.red('generator-botbuilder-java')} generator!`)
-    );
-
-    const prompts = [
-      { name: 'botName', message: `What 's the name of your bot?`, default: 'sample' },
-      { name: 'description', message: 'What will your bot do?', default: 'sample' },
-      { name: 'dialog', type: 'list', message: 'What default dialog do you want?', choices: ['Echo'] },
-    ];
-
-    return this.prompt(prompts).then(props => {
-      // To access props later use this.props.someAnswer;
-      this.props = props;
-    });
-  }
-
-  writing() {
-    const directoryName = _.kebabCase(this.props.botName);
-    const defaultDialog = this.props.dialog.split(' ')[0].toLowerCase();
-
-    if (path.basename(this.destinationPath()) !== directoryName) {
-      this.log(`Your bot should be in a directory named ${directoryName}\nI'll automatically create this folder.`);
-      mkdirp(directoryName);
-      this.destinationRoot(this.destinationPath(directoryName));
-    }
-
-    this.fs.copyTpl(this.templatePath('pom.xml'), this.destinationPath('pom.xml'), { botName: directoryName });
-    this.fs.copy(this.templatePath(`app.java`), this.destinationPath(`app.java`));
-    this.fs.copyTpl(this.templatePath('README.md'), this.destinationPath('README.md'), {
-      botName: this.props.botName, description: this.props.description
-    });
-  }
-
-  install() {
-    this.installDependencies({bower: false});
-  }
+    constructor(args, opts) {
+        super(args, opts);
+
+        // allocate an object that we can use to store our user prompt values from our askFor* functions
+        this.templateConfig = Object.create(null);
+
+        // configure the commandline options
+        this._configureCommandlineOptions(this);
+    }
+
+    initializing() {
+        // give the user some data before we start asking them questions
+        this.log(`\nWelcome to the Microsoft Java Bot Builder generator v${pkg.version}. `);
+    }
+
+    prompting() {
+        const userPrompts = this._getPrompts();
```

Comment:
Iteration order of objects is not guaranteed - it looks like the keys in the object returned by `this._getPrompts()` are unused? Consider using an array instead of an object.